lib/ETL/Yertl.pm

```diff
@@ -20,7 +20,7 @@ sub loop;
 sub import {
     $_[0]->export_to_level( 1, undef, @EXPORT );
     for my $i ( 1..$#_ ) {
-        if ( grep { $_ eq $_[ $i ] } @EXPORT_OK ) {
+        if ( grep { $_ eq ($_[ $i ] //= '') } @EXPORT_OK ) {
```

Comment:
It looks like the problem here is that, inside this loop, we remove stuff from the array with `splice`. But we've already determined how many items we're going to loop over using `for my $i ( 1..$#_ )`. So, then if we remove an item from the `@_` array, we will always generate a warning (since we will loop past the end of the modified `@_` array).

We need to remove items from `@_` because it's going to be used by the method we later `goto` (and we need to use `goto` there to keep the stack how it is), so perhaps instead this block needs to be a C-style for loop (`for ( my $i = 1; $i < $#_; $i++ ) { ... }`) to make sure we keep checking `$#_` as it changes (and not just once at the start).

I think if we use the C-style for loop, it makes it more clear what we're doing (running through the array and removing bits) and avoids re-creating `@_` entries that we don't need anymore.

(side note: This is an attempt to get Import::Base working as an Exporter. Once all the bugs are worked out, I'll likely move the code into Import::Base itself so everything works)

Reply:
Well, I was focused on reducing the number of warnings thrown during ```make test```.  It took me a couple of hours to nail that one, but I concede that I couldn't completely follow the logic of the code in ```import()```.

Note that the other 2 commits in the p.r. are independent of that one, but eliminate ```make test``` warnings as well.

Reply:
Yep. Eliminating warnings is good. I can take this one and fix the bug here if you'd like. Just didn't want to take the opportunity away if you wanted to fix it (I hate when people do that to me).

Reply:
I'll leave that to you as well, as I have other fish to fry.